flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGate.java

```diff
@@ -72,7 +73,19 @@
 
 	void requestPartitions() throws IOException, InterruptedException;
 
-	BufferOrEvent getNextBufferOrEvent() throws IOException, InterruptedException;
+	/**
+	 * Blocking call waiting for next {@link BufferOrEvent}.
+	 *
+	 * @return {@code Optional.empty()} if {@link #isFinished()} returns true.
+	 */
+	Optional<BufferOrEvent> getNextBufferOrEvent() throws IOException, InterruptedException;
```

Comment:
What do you think about this?

Reply:
I would prefer to leave it as it is for now. I completely agree it's a bad design, however in this PR I'm only documenting this behaviour and changing the `null` to `Optional.empty()`, and because of the extensive scope of this change already as it is, I would prefer to fix this issue later on.

Reply:
👍